### DISCUSSION TITLE
Is there a performance penalty when a class is not found by Composer autoload?

### DISCUSSION THREAD

**Discussion by gskema** [composer/composer#11797] [Q&A / Support] [answered]
Let's say I have:

```php
<?php

use NotFoundClass;

#[NotFoundClass]
class Acme {
}
```

Now `Acme` is within composer class map, but `NotFoundClass` is not because it's specifically not installed - missing (e.g. using `composer` `replace: *`). Will this cause slower performance - e.g. filesystem lookup on each request?

**Comment by Seldaek** [comment, marked as answer]
By default yes, it will try to look up that class once, but if you use [authoritative classmaps](https://getcomposer.org/doc/articles/autoloader-optimization.md#optimization-level-2-a-authoritative-class-maps) then you can skip that.
